Goa generation error for Header

This is off latest goa v2 (69da4d44a5c4a67ed4c7713b0bb1a4a223cefb74). Block of code in a Method generates a syntax error in the generated code:
```                
                HTTP(func() {
                        GET("/")
                        Header("etag:If-None-Match")
                        Response(StatusOK, func() {
                                Header("etag:Etag")
                        })
                        Response(StatusNotModified, func() {
                                Tag("not_modified", "true")
                                Body(Empty)
                        })
                })
```
Generates error:
types.go:238:2: expected ';', found 'IDENT' v
 
In which a newline is missing:
    237 v.Items[j].CreatedBy = unmarshalUserToUser(val.CreatedBy)
    238 }v.Etag = etag

 If the `Header("etag:Etag")` line is commented its fine

--- a/http/codegen/client_body_types_test.go
+++ b/http/codegen/client_body_types_test.go
@@ -43,6 +43,7 @@ func TestBodyTypeInit(t *testing.T) {
 		{"body-user-inner", testdata.PayloadBodyUserInnerDSL, 3, BodyUserInnerInitCode},
 		{"body-path-user-validate", testdata.PayloadBodyPathUserValidateDSL, 2, BodyPathUserValidateInitCode},
 		{"body-primitive-array-user-validate", testdata.PayloadBodyPrimitiveArrayUserValidateDSL, 2, BodyPrimitiveArrayUserValidateInitCode},
+		{"result-body-user", testdata.ResultBodyObjectHeaderDSL, 2, ResultBodyObjectHeaderInitCode},
 	}
 	for _, c := range cases {
 		t.Run(c.Name, func(t *testing.T) {
@@ -106,5 +107,17 @@ func NewMethodUserBodyPathValidateRequestBody(p *servicebodypathuservalidate.Pay
 		A: p.A,
 	}
 	return body
+}
+`
+
+const ResultBodyObjectHeaderInitCode = `// NewMethodBodyObjectHeaderMethodBodyObjectHeaderResultOK builds a
+// "ServiceBodyObjectHeader" service "MethodBodyObjectHeader" endpoint result
+// from a HTTP "OK" response.
+func NewMethodBodyObjectHeaderMethodBodyObjectHeaderResultOK(body *MethodBodyObjectHeaderResponseBody, b *string) *servicebodyobjectheader.MethodBodyObjectHeaderResult {
+	v := &servicebodyobjectheader.MethodBodyObjectHeaderResult{
+		A: body.A,
+	}
+	v.B = b
+	return v
 }
 `

--- a/http/codegen/client_types.go
+++ b/http/codegen/client_types.go
@@ -202,9 +202,9 @@ func {{ .Name }}({{- range .ClientArgs }}{{ .Name }} {{ .TypeRef }}, {{ end }}) 
 		{{- end }}
 		{{- if .ReturnIsStruct }}
 			{{- range .ClientArgs }}
-				{{- if .FieldName -}}
+				{{- if .FieldName }}
 			v.{{ .FieldName }} = {{ if .Pointer }}&{{ end }}{{ .Name }}
-				{{ end }}
+				{{- end }}
 			{{- end }}
 		{{- end }}
 		return {{ if .ReturnTypeAttribute }}res{{ else }}v{{ end }}

--- a/http/codegen/testdata/result_dsls.go
+++ b/http/codegen/testdata/result_dsls.go
@@ -569,6 +569,23 @@ var ResultBodyObjectDSL = func() {
 	})
 }
 
+var ResultBodyObjectHeaderDSL = func() {
+	Service("ServiceBodyObjectHeader", func() {
+		Method("MethodBodyObjectHeader", func() {
+			Result(func() {
+				Attribute("a", String)
+				Attribute("b", String)
+			})
+			HTTP(func() {
+				POST("/")
+				Response(StatusOK, func() {
+					Header("b:Authorization")
+				})
+			})
+		})
+	})
+}
+
 var ResultBodyUserDSL = func() {
 	var ResultType = Type("ResultType", func() {
 		Attribute("a", String)

